internal/db/plan.go

```diff
@@ -54,6 +109,48 @@ func GetPlanByID(ctx context.Context, db *gorm.DB, planID string) (*model.Plan,
 	return &plan, nil
 }
 
+// GetActivePlanRate returns the currently active rate for a subscription plan.
+func GetActivePlanRate(ctx context.Context, db *gorm.DB, planID string) (*model.PlanRate, error) {
+	wrapMsg := fmt.Sprintf("unable to look up the active plan rate for '%s'", planID)
+	var err error
+
+	planRate := model.PlanRate{PlanID: &planID}
+	err = db.
+		WithContext(ctx).
+		Where("effective_date <= CURRENT_TIMESTAMP").
+		Order("effective_date desc").
+		Limit(1).
+		Find(&planRate).
+		Error
+
+	// Return the result.
+	if err != nil {
+		return nil, errors.Wrap(err, wrapMsg)
+	}
+	return &planRate, nil
+}
+
+// GetActivePlanQuotaDefaults returns the currently active quota defaults for a subscription plan.
+func GetActivePlanQuotaDefaults(ctx context.Context, db *gorm.DB, planID string) ([]model.PlanQuotaDefault, error) {
+	wrapMsg := fmt.Sprintf("unable to look up the active plan quota defaults for '%s'", planID)
+	var err error
+
+	var planQuotaDefaults []model.PlanQuotaDefault
+	err = db.
+		WithContext(ctx).
+		Select("DISTINCT ON (resource_type_id) resource_type_id", "id", "plan_id", "quota_value", "effective_date").
+		Where("effective_date <= CURRENT_TIMESTAMP AND plan_id = ?", planID).
```

Comment:
This appears to do what I want it to do, but SQL sanity checks are always a good thing. 😄

Reply:
I feel like I'm not quite following what this does. I'd expect from the name for it to return only one default for each resource type in the active plan, but it seems like if there were to be two defaults with different effective dates but both in the past, it'd return both of them. But there's possibly something clever I'm missing, heh.

Reply:
Ah, yes, I see. I confirmed with the PostgreSQL docs and it does explicitly say there that it's the first row and which row is first should be controlled by ORDER BY, so seems perfect.

LGTM now that I wrapped my head around it :D